src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -1619,6 +1672,20 @@ private static bool CleanupReadStateAndCheckPending(State state, ReadState final
             return shouldComplete;
         }
 
+        internal async ValueTask StartAsync(CancellationToken cancellationToken)
+        {
+            Debug.Assert(!Monitor.IsEntered(_state));
+
+            using var registration = cancellationToken.UnsafeRegister((state, token) => {
+                ((State)state!).StartCompletionSource.SetCanceled(token);
+            }, _state);
+
+            uint status = MsQuicApi.Api.StreamStartDelegate(_state.Handle, QUIC_STREAM_START_FLAGS.SHUTDOWN_ON_FAIL | QUIC_STREAM_START_FLAGS.INDICATE_PEER_ACCEPT);
+            QuicExceptionHelpers.ThrowIfFailed(status, "Could not start stream.");
```

Comment:
Do we want to set this exception to TCS too?

Reply:
I don't think this will ever throw, but I added it just in case